commons/src/scanner/symbol/iterator/root.rs

```diff
@@ -47,16 +40,16 @@ impl<'input> Iterator for SymbolIteratorRoot<'input> {
     type Item = &'input Symbol<'input>;
 
     fn next(&mut self) -> Option<Self::Item> {
-        let symbol = self.symbols.get(self.curr_index)?;
+        let symbol = self.symbols.get(self.index)?;
 
-        self.curr_index += 1;
-        self.peek_index = self.curr_index;
+        self.index += 1;
+        self.peek_index = self.index;
 
         Some(symbol)
     }
 
     fn size_hint(&self) -> (usize, Option<usize>) {
-        let len = self.remaining_symbols().unwrap_or(&[]).len();
+        let len = self.remaining_symbols().map(<[_]>::len).unwrap_or(0);
```

Comment:
Hahah, nice! I would say this will be a fun easter egg for future readers. I bet most of the people reading this will wonder what is this and learn something new 🤣

Reply:
😄 it probably will be an easter egg for me in a year, but it looks nice.